showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITPagination.java

```diff
@@ -0,0 +1,167 @@
+        TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testExpandGrpc() {
+    String content = "Testing the entire response is the same";
+    int expected = content.split(" ").length;
+    ServerStream<EchoResponse> echoResponses =
+        grpcClient.expandCallable().call(ExpandRequest.newBuilder().setContent(content).build());
+    List<String> values = new ArrayList<>();
+    for (EchoResponse echoResponse : echoResponses) {
+      values.add(echoResponse.getContent());
+    }
+
+    String response = String.join(" ", values);
+    assertThat(values.size()).isEqualTo(expected);
+    assertThat(response).isEqualTo(content);
+  }
+
+  @Test
+  public void testExpandHttpJson() {
+    String content = "Testing the entire response is the same";
+    int expected = content.split(" ").length;
+    ServerStream<EchoResponse> echoResponses =
+        httpjsonClient
+            .expandCallable()
+            .call(ExpandRequest.newBuilder().setContent(content).build());
+    List<String> values = new ArrayList<>();
+    for (EchoResponse echoResponse : echoResponses) {
+      values.add(echoResponse.getContent());
+    }
+
+    String response = String.join(" ", values);
+    assertThat(values.size()).isEqualTo(expected);
+    assertThat(response).isEqualTo(content);
+  }
+
+  // This tests that pagination returns the correct number of pages + responses and that
+  // the content is correct.
+  //
+  // The pageToken is where the streaming responses come back from and the page size denotes
+  // how many of the responses come back together. i.e for PageSize = 2 and PageToken = 1, see
+  // below:
+  //         | A | Series  | Of  | Words | That  | Will  | Be  | Sent  | Back  | One | By  | One
+  // Page #  | X | 1       | 1   | 2     | 2     | 3     | 3   | 4     | 4     | 5   | 5   | 6
+  // Token # | 0 | 1       | 2   | 3     | 4     | 5     | 6   | 7     | 8     | 9   | 10  | 11
+  @Test
+  public void testPagedExpandWithTokenGrpc() {
+    int pageSize = 2;
+    int pageToken = 1;
+    String content = "A series of words that will be sent back one by one";
+    int contentLength = content.split(" ").length;
+
+    EchoClient.PagedExpandPagedResponse pagedExpandPagedResponse =
+        grpcClient.pagedExpand(
+            PagedExpandRequest.newBuilder()
+                .setContent(content)
+                .setPageSize(pageSize)
+                .setPageToken(String.valueOf(pageToken))
+                .build());
+    String[] expected = content.split(" ");
+    int numResponses = 0;
+    int numPages = 0;
+    for (EchoClient.PagedExpandPage page : pagedExpandPagedResponse.iteratePages()) {
+      for (EchoResponse echoResponse : page.getValues()) {
+        // Add pageToken as offset to the expected array to start indexing at the pageToken
+        assertThat(echoResponse.getContent()).isEqualTo(expected[numResponses + pageToken]);
+        numResponses++;
+      }
+      numPages++;
+    }
+
```

Comment:
Looks like this is used in a couple of places. Can this be moved to a helper method?

Reply:
Sure, created a helper method.